### ISSUE TITLE
Ubuntu 14.04 mkbackup does not backup anything

### ISSUE THREAD

**Issue by Pablohn26** [rear/rear#1989]
#### Relax-and-Recover (ReaR) Issue Template

Fill in the following items before submitting a new issue
(quick response is not guaranteed with free support):

* ReaR version ("/usr/sbin/rear -V"): Relax-and-Recover 2.4 / 2018-06-21

* OS version ("cat /etc/rear/os.conf" or "lsb_release -a" or "cat /etc/os-release"):
```
OS_VENDOR=Ubuntu
OS_VERSION=14.04
```

* ReaR configuration files ("cat /etc/rear/site.conf" and/or "cat /etc/rear/local.conf"):

```
OUTPUT=ISO
BACKUP=NETFS
BACKUP_URL=nfs://213.251.184.109/

```

* Hardware (PC or PowerNV BareMetal or ARM) or virtual machine (KVM guest or PoverVM LPAR): KVM guest

* System architecture (x86 compatible or PPC64/PPC64LE or what exact ARM device): x86_64

* Firmware (BIOS or UEFI or Open Firmware) and bootloader (GRUB or ELILO or Petitboot): GRUB

* Storage (lokal disk or SSD) and/or SAN (FC or iSCSI or FCoE) and/or multipath (DM or NVMe):  virtual local disk

* Description of the issue (ideally so that others can reproduce it): 

When I try to create a backup of my server, there is no information saved on backup.tar.gz. Just a log file named /var/log/rear/rear-hostname.log, that I have attached it. 

[backup.tar.gz](https://github.com/rear/rear/files/2635326/backup.tar.gz)

I am not sure if it is related to #446 

There is no error that suggests the problem. 

* Workaround, if any: no workaround

* Attachments, as applicable ("rear -D mkrescue/mkbackup/recover" debug log files):

```

rear -D mkbackup
Relax-and-Recover 2.4 / 2018-06-21
Using log file: /var/log/rear/rear-comoreclamar.log
Using backup archive '/tmp/rear.TkIQFqVP3JleVJU/outputfs/comoreclamar/backup.tar.gz'
Creating disk layout
Using guessed bootloader 'GRUB' (found in first bytes on /dev/vda)
Creating root filesystem layout
Skipping 'br-ee7d1d7e64de': not bound to any physical interface.
Skipping 'docker0': not bound to any physical interface.
Handling network interface 'eth0'
eth0 is a physical device
Handled network interface 'eth0'
Skipping 'tap0': not bound to any physical interface.
Cannot include keyboard mappings (no keymaps default directory '')
Copying logfile /var/log/rear/rear-comoreclamar.log into initramfs as '/tmp/rear-comoreclamar-partial-2018-12-01T00:36:30+0100.log'
Copying files and directories
Copying binaries and libraries
Copying kernel modules
Copying all files in /lib*/firmware/
Creating recovery/rescue system initramfs/initrd initrd.cgz with gzip default compression
Created initrd.cgz with gzip default compression (115543053 bytes) in 31 seconds
Making ISO image
Wrote ISO image: /var/lib/rear/output/rear-comoreclamar.iso (117M)
Copying resulting files to nfs location
Saving /var/log/rear/rear-comoreclamar.log as rear-comoreclamar.log to nfs location
Creating tar archive '/tmp/rear.TkIQFqVP3JleVJU/outputfs/comoreclamar/backup.tar.gz'
Preparing archive operationOK
Exiting rear mkbackup (PID 9947) and its descendant processes
Running exit tasks
You should also rm -Rf /tmp/rear.TkIQFqVP3JleVJU


```

**Comment by jsmeix** [comment]
Where is the "rear -D mkbackup" debug log file?

Do you use btrfs as filesystem?
If yes, do you use btrfs subvolumes?
If yes, you may have to specify them in BACKUP_PROG_INCLUDE, cf.
https://github.com/rear/rear/blob/master/usr/share/rear/conf/examples/SLE12-btrfs-example.conf

**Comment by jsmeix** [comment]
No news is good news.

**Comment by jsmeix** [comment]
@Pablohn26 

your rear-comoreclamar.log contains (except)
```
2019-01-18 15:53:35.912133471 Including backup/NETFS/default/400_create_include_exclude_files.sh
2019-01-18 15:53:35.915736961 Entering debugscripts mode via 'set -x'.
+ source /usr/share/rear/backup/NETFS/default/400_create_include_exclude_files.sh
++ is_true no
++ case "$1" in
++ return 1
++ '[' NO '!=' YES ']'
```
which does not match our source code in
https://raw.githubusercontent.com/rear/rear/master/usr/share/rear/backup/NETFS/default/400_create_include_exclude_files.sh
that starts with
```
# Backup all that is explicitly specified in BACKUP_PROG_INCLUDE:
for backup_include_item in "${BACKUP_PROG_INCLUDE[@]}" ; do
    test "$backup_include_item" && echo "$backup_include_item"
done > $TMP_DIR/backup-include.txt

# Implicitly also backup all local filesystems as defined in mountpoint_device
# except BACKUP_ONLY_INCLUDE or MANUAL_INCLUDE is set:
if ! is_true "$BACKUP_ONLY_INCLUDE" ; then
    if [ "${MANUAL_INCLUDE:-NO}" != "YES" ] ; then
```
so that it seems your
usr/share/rear/backup/NETFS/default/400_create_include_exclude_files.sh
is somehow broken.

When one of your ReaR scripts is broken you cannot know
what else could be also broken so that I suggest
to re-install ReaR completely anew.

I would also recommend to try out our current
ReaR upstream GitHub master code as follows:

Basically "git clone" our current ReaR upstream GitHub master code
into a separated directory and then configure and run ReaR
from within that directory like:
```
# git clone https://github.com/rear/rear.git

# mv rear rear.github.master

# cd rear.github.master

# vi etc/rear/local.conf

# usr/sbin/rear -D mkbackup
```
Note the relative paths "etc/rear/" and "usr/sbin/".

I recommend to try out our current ReaR upstream GitHub master code
because that is the only place where we fix bugs - i.e. bugs in older
ReaR versions are not fixed by us (i.e. by ReaR upstream).
If it works with our current ReaR upstream GitHub master code
we would appreciate an explicit positive feedback.

**Comment by jsmeix** [comment]
@Pablohn26 
my previous comment
https://github.com/rear/rear/issues/1989#issuecomment-455596494
is probably wrong because by default in usr/share/rear/conf/default.conf
```
BACKUP_PROG_INCLUDE=( )
```
and `set -x` shows nothing when a `for` loop is run with empty argument
```
# set -x
# BACKUP_PROG_INCLUDE=( )
# set -x
# for backup_include_item in "${BACKUP_PROG_INCLUDE[@]}" ; do echo "'$backup_include_item'" ; done )
```
so that I need to do some more analysis
what the actual root cause could be...

**Comment by jsmeix** [comment]
Found it:

@Pablohn26 
your rear-comoreclamar.log contains (excepts)
```
+ source /usr/share/rear/layout/save/GNU/Linux/230_filesystem_layout.sh
...
++ docker_is_running=
++ service docker status
++ docker_is_running=yes
...
++ is_true yes
++ case "$1" in
++ return 0
+++ docker info
+++ grep 'Docker Root Dir'
+++ awk '{print $4}'
++ docker_root_dir=
++ Log '/dev/vda1 is mounted below  (mount point / is under docker control), skipping.'
+++ date '+%Y-%m-%d %H:%M:%S.%N '
++ local 'timestamp=2019-01-18 15:51:06.300003384 '
++ test 1 -gt 0
++ echo '2019-01-18 15:51:06.300003384 /dev/vda1 is mounted below  (mount point / is under docker control), skipping.'
2019-01-18 15:51:06.300003384 /dev/vda1 is mounted below  (mount point / is under docker control), skipping.
++ echo /
++ grep -q '^'
++ continue
```

The matching code is in
usr/share/rear/layout/save/GNU/Linux/230_filesystem_layout.sh
here from our current one in
https://raw.githubusercontent.com/rear/rear/master/usr/share/rear/layout/save/GNU/Linux/230_filesystem_layout.sh
the code excerpts that match the above log lines
```
# docker daemon mounts file systems for its docker containers
# see also https://docs.docker.com/storage/storagedriver/device-mapper-driver/#configure-direct-lvm-mode-for-production
# As it is for container usage only we do not to backup these up or recreate as this disk device is completely under
# control by docker itself (even the recreation of it incl, the creation of the volume group). Usually this is
# done via a kind of cookbook (Chef, puppet or ansible)
docker_is_running=""
service docker status >/dev/null 2>&1 && docker_is_running="yes"
...
        # docker specific exclude part
        if is_true $docker_is_running ; then
            # docker daemon/service is running
            docker_root_dir=$( docker info 2>/dev/null | grep 'Docker Root Dir' | awk '{print $4}' )
            # If $docker_root_dir is in the beginning of the $mountpoint string then FS is under docker control
            # and we better exclude from saving the layout,
            # see https://github.com/rear/rear/issues/1749
            Log "$device is mounted below $docker_root_dir (mount point $mountpoint is under docker control), skipping."
            echo "$mountpoint" | grep -q "^$docker_root_dir" && continue
        fi
```

What looks suspicious in your case is that
it seems ReaR fails to get the `docker_root_dir`
because that is empty in your case.

As a consequence the filesystem that is mounted at the '/' directoy
gets excluded from var/lib/rear/layout/disklayout.conf
so that you have no entry for your root filesystem in your disklayout.conf
which would let `rear recover` fail anyway.

Another consequence is that you don't get automatically the directory
where your root filesystem is mounted (i.e. the `/` directory) included
in your `TMP_DIR/backup-include.txt` file via
usr/share/rear/backup/NETFS/default/400_create_include_exclude_files.sh
so that subsequently nothing appears in your backup via
usr/share/rear/backup/NETFS/default/500_make_backup.sh
that basically calls `tar -c $(cat backup-include.txt)`
to get things included in the backup, cf. the BACKUP_PROG_INCLUDE
description in usr/share/rear/conf/default.conf

@Pablohn26 
do you use docker?

**Comment by jsmeix** [comment]
@Pablohn26 
what does the command `docker info` (run it as root)
output on your system (wee need the whole exact output)?

**Comment by gdha** [comment]
@jsmeix IMHO it is fine to close this issue - the fix will work
